doc/install/install-ceph-gateway.rst

```diff
@@ -1,255 +1,583 @@
 
-#. Load ``mod_proxy_fcgi`` module.
+ ceph-deploy purge <gateway-node1> [<gateway-node2>]
+ ceph-deploy purgedata <gateway-node1> [<gateway-node2>]
 
-   Execute::
+If you execute ``purge``, you must re-install Ceph.
 
-		sudo a2enmod proxy_fcgi
+Change the Default Port
+-----------------------
 
-#. Start Apache service::
+Civetweb runs on port ``7480`` by default. To change the default port (e.g,. to
+port ``80``), modify your Ceph configuration file in the working directory of
+your administration server. Add a section entitled
+``[client.rgw.<gateway-node>]``, replacing ``<gateway-node>`` with the short
+node name of your Ceph Object Gateway node (i.e., ``hostname -s``).
 
-	sudo service apache2 start
+.. note:: In version 0.94, the Ceph Object Gateway does not support SSL. You
+          may setup a reverse proxy web server with SSL to dispatch HTTPS 
+          requests as HTTP requests to CivetWeb.
 
-RPM-based distros
------------------
+For example, if your node name is ``gateway-node1``, add a section like this
+after the ``[global]`` section::
 
-#. Open the ``httpd.conf`` file::
+ [client.rgw.gateway-node1]
+ rgw_frontends = "civetweb port=80"
 
-	sudo vim /etc/httpd/conf/httpd.conf
+.. note:: Ensure that you leave no whitespace between ``port=<port-number>`` in
+          the ``rgw_frontends`` key/value pair. The ``[client.rgw.gateway-node1]``
+          heading identifies this portion of the Ceph configuration file as 
+          configuring a Ceph Storage Cluster client where the client type is a Ceph
+          Object Gateway (i.e., ``rgw``), and the name of the instance is )
+          ``gateway-node1``.
 
-#. Uncomment ``#ServerName`` in the file and add the name of your server. Provide
-   the fully qualified domain name of the server machine
-   (e.g., ``hostname -f``)::
+Push the updated configuration file to your Ceph Object Gateway node
+(and other Ceph nodes)::
 
-	ServerName {fqdn}
+ ceph-deploy --overwrite-conf config push <gateway-node> [<other-nodes>]
 
-#. Update ``/etc/httpd/conf/httpd.conf`` to load ``mod_proxy_fcgi`` module. Add
-   the following to the file::
+To make the new port setting take effect, restart the Ceph Object
+Gateway::
 
-	<IfModule !proxy_fcgi_module>
-	LoadModule proxy_fcgi_module modules/mod_proxy_fcgi.so
-	</IfModule>
+ sudo systemctl restart ceph-radosgw.service
 
-#. Edit the line ``Listen 80`` in ``/etc/httpd/conf/httpd.conf`` with the public
-   IP address of the host that you are configuring as a gateway server. Write
-   ``Listen {IP ADDRESS}:80`` in place of ``Listen 80``.
+Finally, check to ensure that the port you selected is open on the node's
+firewall (e.g., port ``80``). If it is not open, add the port and reload the
+firewall configuration. For Red Hat Enterprise Linux execute::
 
-#. Start httpd service
+  sudo firewall-cmd --list-all
+  sudo firewall-cmd --zone=public --add-port 80/tcp --permanent
+  sudo firewall-cmd --reload
 
-   Execute::
+For Ubuntu, execute::
 
```

Comment:
Possibly you could say if you use iptables as a firewall, execute? (Since other distros may use iptables as firewall as well?)